### PR TITLE
add api usage documentation

### PR DESCRIPTION
**Pull request by nicolasThal**

### CODE REVIEW

File: en/developer_guide/api_transformer.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
+API transformer
+===============
+
+Context
+-------
+
+To decouple the data exposed through the API and the data stored in the model,
+we have decide to use the `Facade design pattern`_.
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: Open Orchestra uses

File: en/developer_guide/api_transformer.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+If you want to add a new entity to the project, and you will expose it through the API,
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: and you need to expose it

File: en/developer_guide/api_transformer.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        /**
+         * @var string
+         */
+        public $bar;
+    }
+
+Facade
+~~~~~~
+
+The facade is the object that will be exposed through the API, let's also say that you
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: through the API. Let's also say you need

File: en/developer_guide/api_transformer.rst [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+need to put the value of ``Foo::bar`` inside a ``baz`` propety.
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: property

File: en/developer_guide/api_transformer.rst [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+
+    class FooFacade implements FacadeInterface
+    {
+        /**
+         * @Serializer\Type("string")
+         */
+        public $baz;
+    }
+
+The ``FooFacade`` class should implement the ``FacadeInterface`` to be reconized as
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: recognized

File: en/developer_guide/api_transformer.rst [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+Transformer
+~~~~~~~~~~~
+
+As the transformation for the ``bar`` to ``baz`` property is not intuitive, you will
+need to create a Transformer to perform the modification.
+
+Create the ``FooTransformer``:
+
+ * It must implement the ``OpenOrchestra\BaseApi\Transformer\TransformerInterface``
+ * Be registered with the tag ``open_orchestra_api.transformer.strategy``
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: Be registered as a service with

File: en/developer_guide/api_transformer.rst [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    use FooBundle\Facade\FooFacade;
+
+    class FooTransformer extends AbstractTransformer
+    {
+        public function getName()
+        {
+            return 'foo';
+        }
+
+        public function tranform($foo)
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: transform

File: en/developer_guide/api_transformer.rst [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+And the declaration :
+
+.. code-block:: yaml
+
+    foo.transformer.foo:
+        class: FooBundle\Transformer\FooTransformer
+        tags:
+            - { name: open_orchestra_api.transformer.strategy }
+
+To limit the dependancy injection, we have create a ``TransformerManager`` which will
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: dependency, Open Orchestra provides a TransformerManager

File: en/developer_guide/api_transformer.rst [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+known all the transformers. This way, you will be able to directly call an other
+transformer with the method ``$this->getTransformer('foo')``.
+
+You will also have access to the ``Router`` and to the ``GroupContext``
+(see `group context page`_)
+
+Usage
+~~~~~
+
+In the controller, to use the transformer, you need to get it through the
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: Maybe something more fluent like 'In the controller, you can accesss your transformer via the TransformerManager'

**Comment by nicolasThal** [comment]
@itkg-ngilain pr updated

File: en/developer_guide/api_transformer.rst [hunk 10]
```diff
@@ -0,0 +1,133 @@
+API transformer
+===============
+
+Context
+-------
+
+To decouple the data exposed through the API and the data stored in the model,
+Open Orchestra uses the `Facade design pattern`_.
+
+If you want to add a new entity to the project, and you need to expose it through the API,
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: to your project and need

File: en/developer_guide/api_transformer.rst [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+And the declaration :
+
+.. code-block:: yaml
+
+    foo.transformer.foo:
+        class: FooBundle\Transformer\FooTransformer
+        tags:
+            - { name: open_orchestra_api.transformer.strategy }
+
+To limit the dependency, Open Orchestra provides a ``TransformerManager`` which will
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: a TransformerManager knowing all the ...

File: en/developer_guide/api_transformer.rst [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+known all the transformers. This way, you will be able to directly call an other
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: you are able

File: en/developer_guide/api_transformer.rst [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+transformer with the method ``$this->getTransformer('foo')``.
+
+You will also have access to the ``Router`` and to the ``GroupContext``
```
Comment (ngilain) [inline en/developer_guide/api_transformer.rst]: You also have access to

**Comment by ngilain** [comment]
:+1: with my 4 little modifications

**Pull request merged**